python-package/lightgbm/basic.py

```diff
@@ -961,14 +961,13 @@ def predict(
             predict_type = _C_API_PREDICT_LEAF_INDEX
         if pred_contrib:
             predict_type = _C_API_PREDICT_CONTRIB
-        int_data_has_header = 1 if data_has_header else 0
 
         if isinstance(data, (str, Path)):
             with _TempFile() as f:
                 _safe_call(_LIB.LGBM_BoosterPredictForFile(
                     self._handle,
                     _c_str(str(data)),
-                    ctypes.c_int(int_data_has_header),
+                    ctypes.c_int(int(data_has_header)),
```

Comment:
bool is subclass of int, so the bytes representations are the same for 0 and False and for 1 and True, so we don't actually need the int conversion at all.
```suggestion
                    ctypes.c_int(data_has_header),
```

Reply:
iiiinteresting, I didn't know that! You teach me something new about Python every week 😊 

Let's try it.